internal/corazawaf/rule_test.go

```diff
@@ -6,9 +6,33 @@ package corazawaf
 import (
 	"testing"
 
+	"github.com/corazawaf/coraza/v3/macro"
+	"github.com/corazawaf/coraza/v3/types"
 	"github.com/corazawaf/coraza/v3/types/variables"
 )
 
+func TestSecActionMessagePropagationInMatchData(t *testing.T) {
+	r := NewRule()
+	r.Msg, _ = macro.NewMacro("Message")
+	r.LogData, _ = macro.NewMacro("Data Message")
+	r.ID_ = 1
+	// SecActions can have nil Operators
+	r.operator = nil
+	tx := NewWAF().NewTransaction()
+	transformationCache := tx.transformationCache
+	matchdata := r.doEvaluate(types.PhaseLogging, tx, transformationCache)
+	if len(matchdata) != 1 {
+		t.Errorf("expected 1 matchdata, got %d", len(matchdata))
+	}
+
+	if want, have := r.Msg.String(), matchdata[0].Message(); want != have {
+		t.Fatalf("unexpected error: want %q, have %q", want, have)
```

Comment:
is `unexpected error` the right message?

Reply:
thanks updated!